src/API.jl

```diff
@@ -977,4 +977,22 @@ Package(name::AbstractString) = PackageSpec(name)
 Package(name::AbstractString, uuid::UUID) = PackageSpec(name, uuid)
 Package(name::AbstractString, uuid::UUID, version::VersionTypes) = PackageSpec(name, uuid, version)
 
+
+"""
+    @__VERSION__ -> Union{VersionNumber, Nothing}
+
+Get the `VersionNumber` of the package which expands this macro. If executed outside of a
+package `nothing` will be returned.
+"""
+macro __VERSION__()
+    pkg_dir = pkgdir(__module__)
+
+    if pkg_dir !== nothing
+        project_data = TOML.parsefile(Types.projectfile_path(pkg_dir))
```

Comment:
Maybe it makes sense to re-compile when the version in Project.toml is changed?

```suggestion
        toml_path = Types.projectfile_path(pkg_dir)
        include_dependency(toml_path)
        project_data = TOML.parsefile(toml_path)
```